packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -0,0 +1,1486 @@
+      expect(items[2]).toHaveAttribute('aria-labelledby', within(items[2]).getByText('Three').id);
+      expect(groups[0]).toContainElement(items[2]);
+      expect(items[3]).toHaveAttribute('aria-labelledby', within(items[3]).getByText('Four').id);
+      expect(groups[1]).toContainElement(items[3]);
+      expect(items[4]).toHaveAttribute('aria-labelledby', within(items[4]).getByText('Five').id);
+      expect(groups[1]).toContainElement(items[4]);
+      expect(items[5]).toHaveAttribute('aria-labelledby', within(items[5]).getByText('Six').id);
+      expect(groups[1]).toContainElement(items[5]);
+
+      act(() => {
+        fireEvent.keyDown(combobox, {key: 'ArrowDown', code: 40, charCode: 40});
+        jest.runAllTimers();
+        fireEvent.keyDown(combobox, {key: 'ArrowDown', code: 40, charCode: 40});
+        jest.runAllTimers();
+        fireEvent.keyDown(combobox, {key: 'ArrowDown', code: 40, charCode: 40});
+        jest.runAllTimers();
+      });
+
+      expect(combobox).toHaveAttribute('aria-activedescendant', items[3].id);
+
+      act(() => {
+        fireEvent.keyDown(combobox, {key: 'Enter', code: 13, charCode: 13});
+        jest.runAllTimers();
+      });
+
+      expect(combobox.value).toBe('Four');
+      expect(() => getByRole('listbox')).toThrow();
+      expect(onInputChange).toHaveBeenCalledTimes(1);
+      expect(onInputChange).toHaveBeenCalledWith('Four');
+      expect(onSelectionChange).toHaveBeenCalledTimes(1);
+      expect(onSelectionChange).toHaveBeenCalledWith('4');
+
+      act(() => {
+        triggerPress(button);
+        jest.runAllTimers();
+      });
+
+      listbox = getByRole('listbox');
+      expect(listbox).toBeVisible();
+
+      items = within(listbox).getAllByRole('option');
+      groups = within(listbox).getAllByRole('group');
+      expect(items).toHaveLength(1);
+      expect(groups).toHaveLength(1);
+      expect(items[0]).toHaveTextContent('Four');
+      expect(items[0]).toHaveAttribute('aria-selected', 'true');
+      expect(groups[0]).toContainElement(items[0]);
+      expect(groups[0]).toHaveAttribute('aria-labelledby', getByText('Section Two').id);
+      expect(() => getByText('Section One')).toThrow();
+    });
+
+    it('sections are not valid selectable values', function () {
+      let {getByRole} = renderSectionComboBox({selectedKey: 'section 1'});
+
+      let combobox = getByRole('combobox');
+      let button = getByRole('button');
+      expect(combobox.value).toBe('');
+
+      act(() => {
+        triggerPress(button);
+        jest.runAllTimers();
+      });
+
+      let listbox = getByRole('listbox');
+      expect(listbox).toBeVisible();
+
+      let groups = within(listbox).getAllByRole('group');
+      expect(groups[0]).not.toHaveAttribute('aria-selected');
+
+      expect(() => within(listbox).getAllByRole('img', {hidden: true})).toThrow();
+    });
+  });
+
+  // TODO: write tests for isDisabled/isReadOnly/other storybook stories
```

Comment:
is this in progress?

Reply:
will be in a followup, pull is a bit too chunky